### PR TITLE
ref/9162 - control on altri recapiti fields

### PR DESCRIPTION
**Pull request by Gabriiii99**
## Short description

added check on table "GIà associati", moved a duplicated method, added new method to insert records in the table

## List of changes proposed in this pull request

- implemented steps to insert record in the table "GIà associati" and checks on the table's fields
- moved a method in the correct file
- updated a test with the correct method (from PF to PG)

## How to test

Simply run `mvn clean test -Dcucumber.filter.tags=@TA_visualizzazioneSezioneAltriRecapitiPG` and verify if the test works fine.

- [x] Does this PR require a manual test in AWS CodeBuild?

### CODE REVIEW

**Comment by Gabriiii99** [comment]
Build success on aws

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 1]
```diff
@@ -539,7 +538,32 @@ public void visualizzazioneSezioneAltriRecapiti() {
     public void visualizzazioneSezioneAltriRecapitiPG() {
         vaiInFondoAllaPagina();
         By altriRecapitiSectionBy = By.id("specialContactTitle");
-        getWebDriverWait(5).withMessage(" Non si visualizza correttamente  il titolo della sezione altri recapiti").until(ExpectedConditions.visibilityOfElementLocated(altriRecapitiSectionBy));
+        By titleGiaAssociatiBy = By.xpath("//p[contains(text(), 'Già associati')]");
+        WebElement tableGiaAssociati = driver.findElement(By.xpath("//table[@aria-label='Già associati']"));
+        By modifyButtonBy = By.xpath(".//button[contains(@id, 'modifyContact')]");
+        By deleteButtonBy = By.xpath(".//button[contains(@id, 'cancelContact')]");
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: If there are multiple element with the same id, you can just do a specific select like so: 
`By.xpath("//div[@id='example']//div[@id='sub-example']//button[@id='real-element']")`

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 2]
```diff
@@ -539,7 +538,32 @@ public void visualizzazioneSezioneAltriRecapiti() {
...
+        getWebDriverWait(10).withMessage("Non si visualizza correttamente  il titolo della sezione altri recapiti").until(ExpectedConditions.visibilityOfElementLocated(altriRecapitiSectionBy));
+        getWebDriverWait(10).withMessage("Non si visualizza il titolo della tabella").until(ExpectedConditions.visibilityOfElementLocated(titleGiaAssociatiBy));
+        getWebDriverWait(10).withMessage("Non si visualizza la tabella dei recapiti già associati").until(ExpectedConditions.visibilityOf(tableGiaAssociati));
+        getWebDriverWait(10).withMessage("Bottone modifica non visualizzato e non cliccabile").until(ExpectedConditions.and(
+                ExpectedConditions.visibilityOfElementLocated(modifyButtonBy),
+                ExpectedConditions.elementToBeClickable(modifyButtonBy)));
+        getWebDriverWait(10).withMessage("Bottone elimina non visualizzato e non cliccabile").until(ExpectedConditions.and(
+                ExpectedConditions.visibilityOfElementLocated(deleteButtonBy),
+                ExpectedConditions.elementToBeClickable(deleteButtonBy)));
+        List<WebElement> tableRows = tableGiaAssociati.findElements(By.xpath(".//tbody/tr"));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:548]: Check the previous comment and see if  you can change something

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 3]
```diff
@@ -539,7 +538,32 @@ public void visualizzazioneSezioneAltriRecapiti() {
...
+        for (WebElement row : tableRows) {
+            List<WebElement> columns = row.findElements(By.xpath(".//td"));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:550]: Same with the above

**Review by afus0 — CHANGES_REQUESTED** [review]

File: src/test/resources/feature/3-destinatario/personaGiuridica/altriRecapitiPG/008_071_visualizzazioneAltriRecapitiPG.feature [hunk 4]
```diff
@@ -16,5 +16,6 @@ Feature: La persona giuridica visualizza tutti gli elementi della sezione altri
     And Nella pagina I Tuoi Recapiti si recupera il codice OTP tramite chiamata request "personaGiuridica"
```
Comment (leleOFA) [inline src/test/resources/feature/3-destinatario/personaGiuridica/altriRecapitiPG/008_071_visualizzazioneAltriRecapitiPG.feature]: delete in line 13 yaml logic

Comment (Gabriiii99) [reply src/test/resources/feature/3-destinatario/personaGiuridica/altriRecapitiPG/008_071_visualizzazioneAltriRecapitiPG.feature]: 16 or 13?

**Review by afus0 — APPROVED** [review]

**Pull request merged**